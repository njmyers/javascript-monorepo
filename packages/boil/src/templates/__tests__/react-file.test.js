import reactFile from "../react-file";

describe("it works", () => {
  test("it is a function ", () => {
    expect(typeof reactFile).toBe("function");
  });

  test("it runs without crashing", () => {
    expect(reactFile("test-component")).toMatchSnapshot();
  });
});
