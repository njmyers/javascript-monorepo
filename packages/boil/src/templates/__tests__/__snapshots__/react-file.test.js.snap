// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it runs without crashing 1`] = `
"/** @flow */
import * as React from 'react'
// types
import type { Props, State } from './types'
// styles
import './test-component.sass'

class TestComponent extends React.Component<Props, State> {
  state = {}

  static defaultProps = {}

  componentDidMount() {
    return null;
  }

  render() {
    return (
      <section className=\\"testComponent\\">
        TestComponent
        <pre>{JSON.stringify(this.props, null, 2)}</pre>
      </section>
    )
  }
}

export default TestComponent
"
`;
